src/detect-mqtt-subscribe-topic.c

```diff
@@ -82,7 +82,7 @@ static InspectionBuffer *MQTTSubscribeTopicGetData(DetectEngineThreadCtx *det_ct
             0) {
         return NULL;
     }
-
+    buffer->inspect = data;
```

Comment:
these are wrong. I will solve this issue differently, no action on your side required

Reply:
Got it -- I just added this to avoid a different issue pointed out by @catenacyber